libs/vkd3d/resource.c

```diff
@@ -3109,26 +3098,38 @@ static HRESULT d3d12_resource_bind_sparse_metadata(struct d3d12_resource *resour
         goto cleanup;
     }
 
-    if ((vr = VK_CALL(vkQueueBindSparse(vk_queue, 1, &bind_info, VK_NULL_HANDLE))) < 0)
+    sparse->init_timeline_value = ++vkd3d_queue->submission_timeline_count;
```

Comment:
This is flawed I think. Sparse commands run completely out of order with normal submissions, so signal ordering is a problem here. I think we need a wait for previous value to be safe.

Reply:
I suppose this is only really an issue if we don't have a dedicated sparse binding queue, but added an unconditional wait nonetheless since it shouldn't harm anything in that case either.

Reply:
Should be fixed now via a dedicated timeline.